File: php/src/Model/Discount.php
<?php

declare(strict_types=1);

namespace Supermarket\Model;

class Discount
{
    public function __construct(
        private Product $product,
        private string $description,
        private float $discount
    ) {
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    public function getDiscountAmount(): float
    {
        return $this->discount;
    }

    public function getProduct(): Product
    {
        return $this->product;
    }
}

File: java/src/main/java/dojo/supermarket/model/Discount.java
package dojo.supermarket.model;

public class Discount {

    private final String description;
    private final double discountAmount;
    private final Product product;

    public Discount(Product product, String description, double discountAmount) {
        this.product = product;
        this.description = description;
        this.discountAmount = discountAmount;
    }

    public String getDescription() {
        return description;
    }

    public double getDiscountAmount() {
        return discountAmount;
    }

    public Product getProduct() {
        return product;
    }
}
